### PR TITLE
[v0.7.3]support MTP in deepseek w8a8 quant model

### PR DESCRIPTION
**Pull request by mengwei805**
### What this PR does / why we need it?
Add support MTP in deepseek w8a8 quant model.

### Does this PR introduce _any_ user-facing change?
1. The quantized MTP layer of deepseek on the current NPU msmodelslim is not quantized, So the MTP layer in deepseek w8a8 quantization weight is still in bflaot16 format;
2. The description file generated by the current msmodelslim tool does not have MTP layer information. Please manually add it to `quantization_config` in `config.json` and set the value to `FLOAT`.

### How was this patch tested?
local tested

### CODE REVIEW

File: vllm_ascend/models/deepseek_mtp.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        attn_metadata: AttentionMetadata,
+        previous_hidden_states: torch.Tensor,
+        inputs_embeds: Optional[torch.Tensor] = None,
+        spec_step_index: int = 0,
+    ) -> torch.Tensor:
+        if inputs_embeds is None:
+            inputs_embeds = self.embed_tokens(input_ids)
+        assert inputs_embeds is not None
+        # masking inputs at position 0, as not needed by MTP
+        inputs_embeds = torch.where((positions == 0).unsqueeze(-1),
```
Comment (MengqingCao) [inline vllm_ascend/models/deepseek_mtp.py:83]: I noticed that this masking is done like the following in vllm. why we use `torch.where` here? Is there any benifits?
```
inputs_embeds[positions == 0] = 0
```

**Comment by MengqingCao** [comment]
Overall LGTM, thanks for this! Please cherry-pick this pr to main when all the review comments are addressed.

File: vllm_ascend/models/deepseek_mtp.py (hunk 1 above)
Comment (mengwei805) [reply vllm_ascend/models/deepseek_mtp.py:83]: The original vLLM method cannot be used in torchair, so it is replaced with an equivalent method.
And the writing of `inputs_embeds[positions == 0]` has poor performance on Ascend devices

**Comment by mengwei805** [comment]
> Overall LGTM, thanks for this! Please cherry-pick this pr to main when all the review comments are addressed.
2025.4.21 refresh:
As 429 is too big, so i submitted a separate PR after 429 was merged, it is https://github.com/vllm-project/vllm-ascend/pull/593
old:
This change is expected to be synchronized to the main branch in https://github.com/vllm-project/vllm-ascend/pull/429

File: vllm_ascend/models/deepseek_mtp.py (hunk 1 above)
Comment (MengqingCao) [reply vllm_ascend/models/deepseek_mtp.py:83]: Got it, thanks for this explanation
